### PR TITLE
EGE A04

### PR DESCRIPTION
**Pull request by SergeyZinkovich**

### CODE REVIEW

**Review by ToriRori — APPROVED** [review]

File: EGE/Gen/EGE/A04.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+        self.text = 'Для каждого из перечисленных ниже десятичных чисел построили двоичную запись. ' \
+                    + 'Укажите число, двоичная запись которого содержит ровно %s.' % \
+                    num_text(
+                        count,
+                        ['единица', 'единицы', 'единиц']
+                        if val != 0 else
+                        ['значащий нуль', 'значащих нуля', 'значащих нулей']
+                    )
+        self.correct = 1
```
Comment (tonichoff) [inline EGE/Gen/EGE/A04.py:44]: For what?

File: EGE/Gen/EGE/A04.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from ...Bits import Bits
+from ...Russian.NumText import num_text
+
+
+class SumNumbers(SingleChoice):
+    def generate(self):
+        av = self.rnd.in_range(17, 127)
+        bv = self.rnd.in_range(17, 127)
+        r = av + bv
+        atext = Bin.hex_text(av)
```
Comment (tonichoff) [inline EGE/Gen/EGE/A04.py]: In original use `hex_or_oct`. Why we now using only _hex_?

File: EGE/Gen/EGE/A04.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        btext = Bin.hex_text(bv)
```
Comment (tonichoff) [inline EGE/Gen/EGE/A04.py]: Here too

**Review by tonichoff — COMMENTED** [review]
Have several questions

File: EGE/Gen/EGE/A04.py (hunk 1 above)
Comment (SergeyZinkovich) [reply EGE/Gen/EGE/A04.py:44]: Cause i want to

File: EGE/Gen/EGE/A04.py (hunk 2 above)
Comment (tonichoff) [reply EGE/Gen/EGE/A04.py]: In personal conversation with Sergey we find here a error - need to use `oct_text` instead of 'hex_text'

**Review by tonichoff — APPROVED** [review]

**Pull request merged**